src/initializer/data_provider.h

```diff
@@ -35,64 +34,80 @@ namespace initializer
         using type = std::function<double(double, double, double)>;
     };
 
-    template<typename ReturnType, std::size_t dim>
-    struct VectorFunctionHelper
-    {
-    };
 
-    template<>
-    struct VectorFunctionHelper<std::array<double, 3>, 1>
-    {
-        using type = std::function<std::array<double, 3>(double)>;
-    };
+    template<std::size_t dim>
+    using ScalarFunction = typename ScalarFunctionHelper<double, dim>::type;
 
-    template<>
-    struct VectorFunctionHelper<std::array<double, 3>, 2>
-    {
-        using type = std::function<std::array<double, 3>(double, double)>;
-    };
 
-    template<>
-    struct VectorFunctionHelper<std::array<double, 3>, 3>
-    {
-        using type = std::function<std::array<double, 3>(double, double, double)>;
-    };
 
     template<std::size_t dim>
-    using ScalarFunction = typename ScalarFunctionHelper<double, dim>::type;
+    using PHAREDict = cppdict::Dict<int, double, std::size_t, std::string, ScalarFunction<dim>>;
 
-    template<std::size_t dim>
-    using VectorFunction = typename VectorFunctionHelper<std::array<double, 3>, dim>::type;
 
 
 
-    template<std::size_t dim>
-    using PHAREDict = cppdict::Dict<int, double, std::size_t, std::string, ScalarFunction<dim>,
-                                    VectorFunction<dim>>;
+    class PHAREDictHandler
```

Comment:
A private default constructor is perhaps a good idea so it blocks accidental non static access.